pkg/apis/core/validation/validation.go

```diff
@@ -4217,11 +4201,8 @@ func ValidateServiceUpdate(service, oldService *core.Service) field.ErrorList {
 		if oldService.Spec.Type != core.ServiceTypeExternalName && oldService.Spec.ClusterIP != "" {
 			allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
 		}
-		// notes:
-		// we drop the IPFamily field when the Dualstack gate is off.
-		// once the gate is on, we start assigning default ipfamily according to cluster settings. in other words
-		// though the field is immutable, we allow (onetime) change from nil==> to value
-		if oldService.Spec.IPFamily != nil {
+		// a user is not allowed to change the IPFamily field, but they are allowed to nil it out (old clients that don't know about IPFamily)
```

Comment:
> they are allowed to nil it out (old clients that don't know about IPFamily)

doesn't the defaulting populate it based on the clusterIP, so by the time we get here, it is set again?

Reply:
https://github.com/kubernetes/kubernetes/blob/c9003a268dff6700506929b247847341ea4d5b33/pkg/apis/core/v1/defaults.go#L136-L160

Reply:
I think the bug was that the validation requiring the field was stricter than the defaulting logic that populates the field if absent

Reply:
after discussion we identified it's "other services"